### PR TITLE
AppCenter Implementation

### PR DESCRIPTION
**Pull request by IsaacHardy**
This connect's our Master builds to AppCenter so we can have CI for all merged pull requests.

AppCenter allows for:
* Automatic builds
* App Signing
* App Deployments
* App Testing
* App Distribution

### CODE REVIEW

File: android/app/src/main/java/host/exp/exponent/generated/DetachBuildConstants.java [hunk 1]
```diff
@@ -3,6 +3,6 @@
 // This file is auto-generated. Please don't rename!
 public class DetachBuildConstants {
 
-  public static final String DEVELOPMENT_URL = "expba3ad9e96a6043fc95726c73cb381a27://t3-pit.newspring-builds.apollos.exp.direct:80";
+  public static final String DEVELOPMENT_URL = "expba3ad9e96a6043fc95726c73cb381a27://192.168.1.73:80";
```
Comment (richarddubay) [inline android/app/src/main/java/host/exp/exponent/generated/DetachBuildConstants.java]: Is this left over from your local build? I remember Dan had something like this change a couple of PR's back and he had to change it back.

File: ios/newspring.xcodeproj/xcshareddata/xcschemes/newspring.xcscheme [hunk 2]
```diff
@@ -43,10 +56,9 @@
       </AdditionalOptions>
    </TestAction>
    <LaunchAction
-      buildConfiguration = "Debug"
+      buildConfiguration = "Release"
```
Comment (richarddubay) [inline ios/newspring.xcodeproj/xcshareddata/xcschemes/newspring.xcscheme]: Should this be concerning?

File: index.ios.js [hunk 3]
```diff
@@ -0,0 +1,4 @@
+import { AppRegistry } from 'react-native';
+import App from './App';
+
+AppRegistry.registerComponent('App', () => App);
```
Comment (richarddubay) [inline index.ios.js]: Do we need to be this specific with `.ios` and `.android` files? They do the same thing.

File: android/app/src/main/assets/appcenter-config.json [hunk 4]
```diff
@@ -0,0 +1,3 @@
+{
+    "app_secret": ""
+}
```
Comment (richarddubay) [inline android/app/src/main/assets/appcenter-config.json]: There is almost nothing here ... is this necessary?

**Comment by richarddubay** [comment]
@IsaacHardy Also, I noticed that the tests failed because something to do with Coveralls. Shrug.

File: android/app/src/main/assets/appcenter-config.json (hunk 4 above)
Comment (IsaacHardy) [reply android/app/src/main/assets/appcenter-config.json]: I do not know what this is... I'll investigate

File: index.ios.js (hunk 3 above)
Comment (IsaacHardy) [reply index.ios.js]: We absolutely need them

File: ios/newspring.xcodeproj/xcshareddata/xcschemes/newspring.xcscheme (hunk 2 above)
Comment (IsaacHardy) [reply ios/newspring.xcodeproj/xcshareddata/xcschemes/newspring.xcscheme]: Nope. When appcenter builds, we want it to use Release settings

File: android/app/src/main/java/host/exp/exponent/generated/DetachBuildConstants.java (hunk 1 above)
Comment (IsaacHardy) [reply android/app/src/main/java/host/exp/exponent/generated/DetachBuildConstants.java]: It's quite possible. I'll investigate to make sure.

File: index.ios.js (hunk 3 above)
Comment (IsaacHardy) [reply index.ios.js]: The build scripts looks for the index file based on scope (e.g. web, ios, android). So we have to be a bit redundant to make sure AppCenter only pulls mobel builds.

**Pull request merged**